daemon/daemon.go

```diff
@@ -189,6 +191,21 @@ func (daemon *Daemon) Config() config.Config {
 	return daemon.config().Config
 }
 
+// engineSocket returns the first Unix socket from the daemon's configured hosts.
+// Returns an error if no Unix socket is configured.
+func (daemon *Daemon) engineSocket(_ string, opts *mount.APISocketOptions) (string, error) {
+	if opts != nil && opts.Isolation != mount.IsolationNone {
+		return "", fmt.Errorf("isolation %q is not supported", opts.Isolation)
+	}
+	cfg := daemon.Config()
+	for _, host := range cfg.Hosts {
+		if proto, addr, ok := strings.Cut(host, "://"); ok && proto == "unix" {
+			return addr, nil
+		}
+	}
+	return "", errors.New("no unix socket configured in daemon hosts")
```

Comment:
Considering if we should always create a socket for this, but it could be a separate one, e.g. somewhere under `/var/lib/docker`, or some path under the runtime dir. That way we can provide this feature regardless of how the daemon is configured, and it would be "closer" to the per-container socket.